src/interaction/interactive_api.jl

```diff
@@ -177,15 +177,22 @@ function screen_relative(scene::Scene, mpos)
 end
 
 """
-    mouseposition(scene = hovered_scene())
+    mouseposition(ax/scene)
 
 Return the current position of the mouse in _data coordinates_ of the
-given `scene`.
-
-By default uses the `scene` that the mouse is currently hovering over.
+given object.
 """
 mouseposition(x) = mouseposition(get_scene(x))
 
+function mouseposition(ax::Axis)
+    pos = mouseposition(get_scene(ax))
+
+    # `pos` has the axis scaling already applied to it, so to get the true data
+    # coordinates we have to invert the scaling.
+    return Vec2{Float64}(inverse_transform(ax.xscale[])(pos[1]),
+                         inverse_transform(ax.yscale[])(pos[2]))
```

Comment:
This should probably be:

```julia
return Vec2{Float64}(Makie.apply_transform(Makie.inverse_transform(ax.scene.transformation.transform_func[]), pos))
```

but you'd have to test that.

Reply:
Yea there is also the question of "Why just Axis?". Transform functions can be part of any Scene and technically also be different per plot. And if it is per plot it becomes unclear to me what that space even refers to on a Scene global thing like mouseposition()

Reply:
Yep that implementation works, changed in 4abb55536.

> Yea there is also the question of "Why just Axis?". Transform functions can be part of any Scene and technically also be different per plot. And if it is per plot it becomes unclear to me what that space even refers to on a Scene global thing like mouseposition()

I don't think it has to be just Axis, but that's the one that I as a user care most about it.